src/InstrumentationEngine.ProfilerProxy/dllmain.cpp

```diff
@@ -112,8 +113,19 @@ namespace ProfilerProxy
         WCHAR wszEngineFullPath[MAX_PATH];
         IfFailRetErrno_Proxy(wcscpy_s(wszEngineFullPath, MAX_PATH, wszProfilerPath));
 
-        IfFailRet_Proxy(PathCchAppend(wszEngineFullPath, MAX_PATH, wszVersionFolder));
-        IfFailRet_Proxy(PathCchAppend(wszEngineFullPath, MAX_PATH, profilerRelativeFileName));
+        if (!wcslen(wszEngineFullPath) + wcslen(wszVersionFolder) + wcslen(profilerRelativeFileName) < MAX_PATH - 1||
```

Comment:
I assume you are doing MAX_PATH - 1 to allow the null terminator. I don't think this accounts for the directory separators?